file_process/h5ad/h5ad_processor.py

```diff
@@ -3,17 +3,40 @@
 
 import anndata
 import pandas as pd
+import numpy as np
+import h5py
 
 from file_process.base import FileProcessorBase
 from file_process.constants import PREVIEW_ROWS_COUNT
 from file_process.h5ad.h5ad_validator import H5ADValidator
 from file_process.h5ad.schemas import SbioModelDataForH5ad
 
 
+def _h5_process(file):
+
+    data_mat = h5py.File(file, 'r')
+    main_df = np.array(data_mat['X'])
+    if 'Y' in data_mat:
+        obs = np.array(data_mat['Y'])
+    else:
+        obs = None
+    data_mat.close()
+
+    # preprocessing scRNA-seq read counts matrix
+    adata = anndata.AnnData(main_df)
+    if obs is not None:
+        adata.obs['Group'] = obs
+
+    return adata
+
+
 class H5ADFileProcessor(FileProcessorBase):
 
-    def __init__(self, file, **_):
-        self.adata = anndata.read_h5ad(file)
+    def __init__(self, file, ext, **_):
+        if ext == 'h5':
+            self.adata = _h5_process(file)
```

Comment:
I think this should just be a different class that will process the h5 files
it may inherit from some base class for h5 group with all the other methods like validate, get_targets and so on